src/merge.rs

```diff
@@ -0,0 +1,138 @@
+use std::marker;
+use std::mem;
+use libc::c_uint;
+
+use {raw, Object, FileFavor};
+use util::Binding;
+use call::Convert;
+
+/// annotated commits, the input to merge and rebase
+pub struct AnnotatedCommit<'repo> {
+    raw: *mut raw::git_annotated_commit,
+    _marker: marker::PhantomData<Object<'repo>>,
```

Comment:
re phantom data: interesting, i'll take a look.  I vaguely recall that you can create annotated commits from things other than commits but I may well be mistaken.